segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/ZooKeeperServiceRunner.java

```diff
@@ -55,6 +82,19 @@ public void initialize() throws IOException {
         if (this.tmpDir.compareAndSet(null, IOUtils.createTempDir("zookeeper", "inproc"))) {
             this.tmpDir.get().deleteOnExit();
         }
+        if (secureZK) {
```

Comment:
Don't we have to set `secureClientPort` somewhere?

https://zookeeper.apache.org/doc/r3.5.4-beta/zookeeperAdmin.html#sc_configuration

Reply:
From the config: `Note that SSL feature will be enabled when user plugs-in zookeeper.serverCnxnFactory, zookeeper.clientCnxnSocket as Netty.` I do not want to expose mixed mode. Only using the secure mode with the default client port is good enough.

Reply:
@fpj We do not depend on the default way to start ZK. We start it ourselves in the `ZooKeeperServiceRunner.run()` method. This approach was already present. This PR  ensures that we configure it in secure mode.
The `waitForServerUp` method ensures that the ZK server comes up with TLS enabled.

Reply:
Yes. This is right. And this way we know that 1. The server is up 2. The port is configured to use SSL.